diff -burNp xfburn-0.4.0/po/LINGUAS xfburn-0.4.0-new/po/LINGUAS
--- xfburn-0.4.0/po/LINGUAS	2008-11-20 08:15:12.000000000 +0100
+++ xfburn-0.4.0-new/po/LINGUAS	2008-12-29 16:53:25.139176105 +0100
@@ -1,2 +1,2 @@
 # set of available languages (in alphabetic order)
-ar ca cs da de dz el en_GB es et eu fi fr gl he hu id it ja ko lt lv nb_NO nl pa pl pt_BR pt_PT ru sq sv tr uk ur zh_CN zh_TW
+ar ca cs da de dz el en_GB es et eu fi fr gl he hu id it ja ko lt lv nb nl pa pl pt_BR pt ru sq sv tr uk ur zh_CN zh_TW
