diff -burNp xfburn-0.4.3/configure xfburn-0.4.3-dud/configure
--- xfburn-0.4.3/configure	2010-02-13 07:43:15.000000000 +0100
+++ xfburn-0.4.3-dud/configure	2010-02-21 19:34:21.880510484 +0100
@@ -11271,7 +11271,7 @@ _ACEOF
 
 
 
-    ALL_LINGUAS="ar ast bg ca cs da de el en_AU en_GB en es et eu fi fr gl he hr hu id is it ja kk ko lt lv nb nl nn oc pa pl pt_BR pt ru si sk sq sr sv th tr ug uk ur_PK ur vi zh_CN zh_TW "
+    ALL_LINGUAS="ar ast bg ca cs da de el en_AU en_GB en es et eu fi fr gl he hr hu id is it ja kk ko lt lv nb nl nn oc pa pl pt_BR pt ru si sk sq sr sv th tr ug uk ur vi zh_CN zh_TW "
 
    for ac_header in locale.h
 do :
diff -burNp xfburn-0.4.3/configure.in xfburn-0.4.3-dud/configure.in
--- xfburn-0.4.3/configure.in	2010-02-13 07:43:05.000000000 +0100
+++ xfburn-0.4.3-dud/configure.in	2010-02-21 19:34:11.173231576 +0100
@@ -31,7 +31,7 @@ AC_PROG_INTLTOOL([0.31], [no-xml])
 AC_SYS_LARGEFILE
 
 dnl check for i18n support
-XDT_I18N([ar ast bg ca cs da de el en_AU en_GB en es et eu fi fr gl he hr hu id is it ja kk ko lt lv nb nl nn oc pa pl pt_BR pt ru si sk sq sr sv th tr ug uk ur_PK ur vi zh_CN zh_TW ])
+XDT_I18N([ar ast bg ca cs da de el en_AU en_GB en es et eu fi fr gl he hr hu id is it ja kk ko lt lv nb nl nn oc pa pl pt_BR pt ru si sk sq sr sv th tr ug uk ur vi zh_CN zh_TW ])
 
 dnl check for required packages
 XDT_CHECK_PACKAGE([LIBBURN], [libburn-1], [0.4.2])
